src/App.tsx

```diff
@@ -15,36 +16,112 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortBy {
+  Alphabetically = 'alphabetically',
+  Length = 'length',
+}
+
+interface SortParams {
+  sortField: string;
+  isReversed: boolean;
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, isReversed }: SortParams,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortBy.Alphabetically:
+          return good1.localeCompare(good2);
+
+        case SortBy.Length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    return preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
```

Comment:
[CODE KNOWLEDGE] - The `getPreparedGoods` function directly manipulates the sort order of the goods, which is a form of direct DOM interaction. Consider using React state and effects to handle sorting and reversing logic, ensuring that all UI updates are handled in a declarative manner by React.